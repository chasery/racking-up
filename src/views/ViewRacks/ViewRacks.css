.ViewRacks {
  padding-top: 4.5rem;
}

.ViewRacks__wrapper {
  padding: 1rem 1.5rem;
}

@media all and (min-width: 50rem) {
  .ViewRacks__wrapper {
    max-width: 64rem;
    align-self: center;
    margin: 0 auto;
  }
}
