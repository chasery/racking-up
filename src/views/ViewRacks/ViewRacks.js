import React, { useState, useEffect } from 'react';
import RacksApiService from '../../services/racks-api-service';
import Header from '../../components/Header/Header';
import RacksList from '../../components/RacksList/RacksList';
import Error from '../../components/Error/Error';
import './ViewRacks.css';

function ViewRacks(props) {
  const [racks, setRacks] = useState([]);
  const [error, setError] = useState(null);

  useEffect(() => {
    async function initState() {
      try {
        let apiCall = await RacksApiService.getRacks();
        let res = await apiCall;

        setRacks(res);
      } catch (error) {
        setError(error.error);
      }
    }

    initState();
  }, []);

  return (
    <>
      <Header />
      <main role='main'>
        <section className='ViewRacks'>
          <div className='ViewRacks__wrapper'>
            {error ? (
              <Error message={error} />
            ) : (
              <>
                <RacksList racks={racks} />
              </>
            )}
          </div>
        </section>
      </main>
    </>
  );
}

export default ViewRacks;
