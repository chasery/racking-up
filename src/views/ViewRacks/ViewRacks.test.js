import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import ViewRacks from "./ViewRacks";

describe("ViewRacks component", () => {
  it("renders without error", () => {
    const wrapper = shallow(<ViewRacks />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
