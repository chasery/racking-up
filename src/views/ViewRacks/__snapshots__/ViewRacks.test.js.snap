// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewRacks component renders without error 1`] = `
<main
  role="main"
>
  <section
    className="ViewRacks"
  >
    <div
      className="ViewRacks__wrapper"
    >
      <h3>
        My Racks
      </h3>
      <RacksList />
    </div>
  </section>
</main>
`;
